packages/admin/cms-admin/src/dependencies/DependencyList.tsx

```diff
@@ -131,79 +167,49 @@ export const DependencyList = ({ query, variables }: DependencyListProps) => {
         },
     ];
 
-    let items: DependencyItem[] = [];
-    let totalCount = 0;
+    const { data, loading, error, refetch } = useQuery<Query, QueryVariables>(query, {
+        variables: {
+            offset: dataGridProps.paginationModel.page * dataGridProps.paginationModel.pageSize,
+            limit: dataGridProps.paginationModel.pageSize,
+            ...variables,
+        },
+    });
+
+    if (error) throw error;
 
-    if (error) {
-        throw error;
+    if (!loading && ((data?.item.dependencies && data?.item.dependents) || (!data?.item.dependents && !data?.item.dependencies))) {
+        throw new Error("Either dependencies or dependents must be defined, but not both.");
     }
 
-    if (data?.item.dependencies) {
-        items = data.item.dependencies.nodes.map((node) => ({
-            ...node,
-            graphqlObjectType: node.targetGraphqlObjectType,
-            id: node.targetId,
-        }));
-        totalCount = data.item.dependencies.totalCount;
-    } else if (data?.item.dependents) {
-        items = data.item.dependents.nodes.map((node) => ({
-            ...node,
-            graphqlObjectType: node.rootGraphqlObjectType,
-            id: node.rootId,
-        }));
-        totalCount = data.item.dependents.totalCount;
-    } else if (!loading) {
-        throw new Error("Neither dependencies nor dependents is defined");
-    }
+    const type: "dependencies" | "dependents" = data?.item.dependencies ? "dependencies" : "dependents";
+
+    const rowCount = useBufferedRowCount(data?.item[type]?.totalCount);
+    const rows =
+        data?.item[type]?.nodes.map((node) => {
+            return {
+                ...node,
+                graphqlObjectType: type === "dependencies" ? (node as Dependency).targetGraphqlObjectType : (node as Dependent).rootGraphqlObjectType,
+                id: type === "dependencies" ? (node as Dependency).targetId : (node as Dependent).rootId,
+            };
+        }) ?? [];
```

Comment:
General question: maybe it would be better to have two separate components DependencyList and DependentList?